runtime/src/bridge.rs

```diff
@@ -86,6 +91,36 @@ decl_module! {
             Ok(())
         }
 
+        /// Transfer an nft to a whitelisted destination chain. Source nft is locked in bridge account
+        /// rather than being burned.
+        #[weight = 195_000_000]
+        pub fn transfer_asset(origin,
+                              recipient: Vec<u8>,
+                              from_registry: RegistryId,
+                              token_id: TokenId,
+                              resource_id: ResourceId,
+                              dest_id: chainbridge::ChainId,
+        ) -> DispatchResult {
+            let source = ensure_signed(origin)?;
+
+            // Chain must be whitelisted
+            ensure!(<chainbridge::Module<T>>::chain_whitelisted(dest_id), Error::<T>::InvalidTransfer);
+
+            // Burn additional fees
+            let nft_fee: T::Balance = NFT_FEE.saturated_into();
+            <fees::Module<T>>::burn_fee(&source, nft_fee)?;
+
+            // Lock asset by transfering to bridge account
+            let bridge_id = <chainbridge::Module<T>>::account_id();
+            let asset_id = AssetId(from_registry, token_id);
+            <nft::Module<T> as Unique>::transfer(&source, &bridge_id, &asset_id)?;
+
+            // Transfer instructions for relayer
+            let tid: &mut [u8] = &mut[0; 32];
+            token_id.to_big_endian(tid);
+            <chainbridge::Module<T>>::transfer_nonfungible(dest_id, resource_id, tid.to_vec(), recipient, vec![]/*assetinfo.metadata*/)
```

Comment:
As mentioned offline, I wanted to throw out there, if we needed to add a list of supported registries per chain_id, so there is more control on where to move the nft. Since as it stands, as soon as we whitelist another nft registry in ethereum, we would be able to "accidentally" use the resource_id of that registry instead of the correspondent one.
I believe that might be duty of the resource_id mapping that the bridge is responsible for, but lets put some thought into it.

Reply:
Ok from my understanding, we need a mapping `m: C -> E`
```
E = set of Tinlake contracts (on Ethereum)
C = set of registries on Centrifuge
```
Not every registry necessarily maps to a contract in `E`, and its possible that multiple registries map to the same contract, so `m` is not injective. But every Tinlake contract in `E` has a corresponding registry in `C` so `m` is surjective.

So a substrate StorageMap `ResourceId => ResourceId` to represent `m` would be sufficient.

We can put this into the centrifuge `src/bridge.rs` and allow access through the bridge admin account. I'm not sure how the proposal process works in the parent chainbridge-pallet and if that can be integrated to?

Reply:
Yes that is correct. TInlake is just a consumer of the NFTs, so we can simplify to:
```
E = set of nft registries on Ethereum
C = set of registries on Centrifuge
```
I believe that it should be duty of the bridge contracts/pallets admin'd by the relayers (chainbridge-pallet and erc721-handler) that is whitelisted by the admin operators. Since eventually, the bridge will keep track of where the assets are moving to/from, sounds like a first step towards it.

A Centrifuge Asset can only be minted in the centrifuge protocol, so lets call it the NFT source of truth, so any move will always originate in the cent chain first `C -> E`. Since we always lock the asset before sending, when the NFT comes back `E -> C` (if it does), a check can be performed to verify that. the token_id exists in the desired registry, and owned by the bridge address, but that sounds like the duty of the centrifuge receiver function in `src/bridge.rs`. 

So a whitelist storage map in both sides `centrifuge-chain` and `ethereum`, and that means that we would need to modify the chainbridge-substrate pallet and the chainbridge erc721-handler contract.
That would mean that we need a new deployment on ethereum, and therefore a cascade of changes around most of the layers, which is a bit unfortunate.
There are some alternative solutions, and I would like put more thought. Let sync offline about it.